src/supremm/datasource/prometheus/promsummarize.py

```diff
@@ -0,0 +1,301 @@
+                if self.fail_fast:
+                    raise
+
+        return success == 0
+
+    def processnode(self, mdata):
+        """ Process a single node from a job """
+
+        start, end = self.job.start_datetime.timestamp(), self.job.end_datetime.timestamp()
+        ctx = Context(start, end, self.mapping.client)
+
+        for preproc in self.preprocs:
+            ctx.mode = preproc.mode
+            self.processforpreproc(ctx, mdata, preproc)
+
+        for analytic in self.alltimestamps:
+            ctx.mode = analytic.mode
+            self.processforanalytic(ctx, mdata, analytic)
+
+        for analytic in self.firstlast:
+            ctx.mode = analytic.mode
+            self.processfirstlast(ctx, mdata, analytic)
+
+    def processforpreproc(self, ctx, mdata, preproc):
+        """ Fetch the data from Prometheus and pass entire response
+            to the preprocessor runcallback function
+        """
+
+        preproc.hoststart(mdata.nodename)
+        logging.debug("Processing %s (%s)" % (type(preproc).__name__, preproc.name))
+
+        reqMetrics = self.mapping.getmetricstofetch(preproc.requiredMetrics)
+        if False == reqMetrics:
+            logging.warning("Skipping %s (%s)." % (type(preproc).__name__, preproc.name))
+            preproc.hostend()
+            return
+
+        results = ctx.fetch(reqMetrics)
+
+        done = False
+        while not done:
+            try:
+                result = next(results)
+                if False == self.runpreproccall(preproc, result, ctx, mdata):
+                    break
+            except StopIteration:
+                done = True
+            except requests.RequestException as exp:
+                analytic.status = "failure"
+                raise exp
+            except Exception as exp:
+                preproc.status = "failure"
+                preproc.hostend()
+                raise exp
+
+        preproc.status = "complete"
+        preproc.hostend()
+
+    def processfirstlast(self, ctx, mdata, analytic):
+        """ Fetch the data from Prometheus and pass entire response
+            to the analytic runcallback function
+        """
+        logging.debug("Processing %s (%s)" % (type(analytic).__name__, analytic.name))
+
+        reqMetrics = self.mapping.getmetricstofetch(analytic.requiredMetrics)
+        if False == reqMetrics:
+            logging.warning("Skipping %s (%s)." % (type(analytic).__name__, analytic.name))
+            analytic.status = "failure"
+            return
+
+        results = ctx.fetch(reqMetrics)
+
+        try:
+            result = next(results)
+        except StopIteration:
+            analytic.status = "failure"
+            return
+        except requests.RequestException as exp:
+            analytic.status = "failure"
+            raise exp
+        except Exception as exp:
+            analytic.status = "failure"
+            raise exp
```

Comment:
```suggestion
        except Exception as exp:
            analytic.status = "failure"
            raise exp
```
These `except` blocks can be combined since `requests.RequestException` inherits from `Exception`.